File: lecture_009(Recursion)/l_013_firstindex.java
import java.util.Scanner;
public class l_013_firstindex {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        int d = scn.nextInt();
        int fi = firstIndex(arr,0,d);
        System.out.println(fi);
    }
    public static void input(int[] arr) {
        for(int i=0; i<arr.length; i++) {
            arr[i] = scn.nextInt();
        }
    }
    public static int firstIndex(int[] arr, int idx, int x) {
        if(idx==arr.length) {
            return -1;
        }
        //Short approach
        if(arr[idx]==x) {
            return idx;
        }
        else {
            int fiisa = firstIndex(arr,idx+1,x);   //fiisa = first index in small array
            return fiisa;
        }
    }
}

File: lecture_009(Recursion)/l_013_firstindex.cpp
#include<iostream>
#include<vector>
using namespace std;

void input(vector<int> &arr) {
    for(int i=0; i<arr.size(); i++) {
        cin>>arr[i];
    }
}
int firstIndex(vector<int> &arr, int idx, int x) {
    if(idx==arr.size()) {
        return -1;
    }
    //Long approach
    int fiisa = firstIndex(arr,idx+1,x);
    if(arr[idx]==x) {
        return idx;
    }
    else {
        return fiisa;
    }
}
int main() {
    int n;
    cin>>n;
    vector<int>arr(n,0);
    input(arr);
    int d;
    cin>>d;
    int fi = firstIndex(arr,0,d);
    cout<<fi<<endl;

    return 0;
}